### PR TITLE
Fix string_view issue breaking suitability window

### LINKED ISSUE #3772: Planet suitability shown as "not suitable" for species already living there
Bug Report
==========

Environment
-----------

* **FreeOrion Version**: fdeac67
* **Operating System**: Ubuntu 20.04
  * Compiled from source

Description
-----------
When player checks out planet suitability of a populated planet owned by them, it is shown as not suitable for colonization for the species already living there:
wrong_target_pop_for_colonizing_own_homeworld

Any other planet suitability value is shown correctly: unpopulated (owned or not), owned by other empires (populated or not).

Expected Result
---------------
It should show target population.

Steps to reproduce
------------------
Quickstart a game and check out planet suitability of empire's capital.

### PR DESCRIPTION
**Pull request by Grummel7**
Fixes #3772

### CODE REVIEW

**Comment by Grummel7** [comment]
Redone completely, fixing it at the source is simpler and more robust.

**Comment by geoffthemedio** [comment]
That was unfortunate timing...
https://github.com/freeorion/freeorion/commit/4e930bce474abc2d9400ff5713d01cdc623008cc

**Comment by geoffthemedio** [comment]
can you rebase on master and explicitly undo the now unnecessary changes in that commit and the following?

**Comment by Grummel7** [comment]
Done it -- I should stop doing late-night PRs...

File: UI/EncyclopediaDetailPanel.cpp [hunk 1]
```diff
@@ -2982,18 +2984,15 @@ namespace {
         ScriptingContext context{universe, Empires(), GetGalaxySetupData(), GetSpeciesManager(), GetSupplyManager()};
         universe.InhibitUniverseObjectSignals(true);
 
-        // planet->SetSpecies modifies a string that may be part of the input
-        // string_view vector, so make copies now to use while iterating
-        std::vector<std::string> names_copy(species_names.begin(), species_names.end());
-        for (auto& species_name : names_copy) { // TODO: parallelize somehow? tricky since an existing planet is being modified, rather than adding a test planet...
```
Comment (geoffthemedio) [inline UI/EncyclopediaDetailPanel.cpp:2988]: if not iterating over `string` but instead `const string_view` then should also remove the `move` on line 3013

File: UI/EncyclopediaDetailPanel.cpp [hunk 2]
```diff
@@ -3009,8 +3009,7 @@ namespace {
                 float planet_capacity = ((planet_environment == PlanetEnvironment::PE_UNINHABITABLE) ?
                                          0.0f : planet->GetMeter(MeterType::METER_TARGET_POPULATION)->Current()); // want value after temporary meter update, so get current, not initial value of meter
 
-                retval.emplace(std::piecewise_construct, std::forward_as_tuple(planet_capacity),
-                               std::forward_as_tuple(std::move(species_name), planet_environment));
+                retval.emplace(planet_capacity, std::pair{species_name, planet_environment});
```
Comment (geoffthemedio) [inline UI/EncyclopediaDetailPanel.cpp]: leave the `piecewise_construct` and `forward_as_tuple`; just remove `std::move()`

**Pull request merged**